configuration/src/main/java/io/airlift/configuration/ConfigurationFactory.java

```diff
@@ -78,7 +79,7 @@ public class ConfigurationFactory
     private static final Validator VALIDATOR;
 
     private static final TypeToken<List<String>> LIST_OF_STRINGS_TYPE_TOKEN = new TypeToken<List<String>>() {};
-    private static final Splitter VALUE_SPLITTER = Splitter.on(",").omitEmptyStrings().trimResults();
+    private static final Splitter VALUE_SPLITTER = Splitter.on(Pattern.compile("(?<!\\\\),")).omitEmptyStrings().trimResults();
```

Comment:
lgtm: object names cannot have `\`'s in them (suggested to use identifier chars), i.e. so no need to worry about multiple escapes, etc.

Reply:
Note that this is changing how parsing is done for all config values of type `List<String>`. It's not limited to just object names.